dht/i_partitioner.cc

```diff
@@ -439,4 +443,44 @@ future<dht::partition_range_vector> subtract_ranges(const schema& schema, const
     co_return res;
 }
 
+dht::token_range_vector split_token_range_msb(unsigned most_significant_bits) {
+    dht::token_range_vector ret;
+    // Avoid shift left 64
+    if (!most_significant_bits) {
+        auto&& start_bound = dht::token_range::bound(dht::minimum_token(), true);
+        auto&& end_bound = dht::token_range::bound(dht::maximum_token(), true);
```

Comment:
nit: I don't understand this `auto&& var = some_type();` This result of constructing an object is a value. Saving a reference to this temporary will result in use-after-free.